### PR TITLE
refactor: console log

### PR DESCRIPTION
**Pull request by nugaon**
`console.log` becomes more structured within this PR.
* `console.log` now only accessible through `root-command` parent class by default 
* some message types have been defined (e.g. `error`, `info`, `dim`, `divider`...)
* auto message formats for message types (italic, bold, etc.) 
* 3 levels of verbosity for restrict output messages (Quiet, Normal, Verbose) with optimized console message calling

Resolves: #13, resolves: #14

### CODE REVIEW

File: src/command/upload.ts [hunk 1]
```diff
@@ -129,11 +136,11 @@ export class Upload extends RootCommand implements LeafCommand {
     syncedBar.stop()
 
     if (!synced) {
-      console.warn(red('Data syncing timeout.'))
+      this.console.error('Data syncing timeout.')
 
       return false
     } else {
-      console.log(dim('Data has been synced on Swarm network'))
+      this.console.log(dim('Data has been synced on Swarm network'))
```
Comment (vojtechsimetka) [inline src/command/upload.ts:143]:
```suggestion
      this.console.dim('Data has been synced on Swarm network')
```

File: src/command/upload.ts [hunk 2]
```diff
@@ -5,8 +5,9 @@ import * as Path from 'path'
 import { sleep } from '../utils'
 import { Tag } from '@ethersphere/bee-js/dist/types'
 import { SingleBar, Presets } from 'cli-progress'
-import { bold, green, dim, red } from 'kleur'
+import { bold, green, dim } from 'kleur'
```
Comment (vojtechsimetka) [inline src/command/upload.ts:8]: These should probably not be used here, right?
```suggestion
```

File: src/command/root-command/command-log.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        break
+      default:
+        // quite
+        this.error = error
+        this.log = emptyFunction
+        this.info = emptyFunction
+        this.dim = emptyFunction
+        this.divider = emptyFunction
+    }
+  }
```
Comment (vojtechsimetka) [inline src/command/root-command/command-log.ts:64]: IMHO there should be `--silent` option as well which really just logs in the important stuff like hash on upload or really nothing on download.

Comment (vojtechsimetka) [reply src/command/root-command/command-log.ts:64]: Or maybe just provide `--log-level` attribute with following options: `error, log, info, dim` or numeric value `1-4`

File: src/command/root-command/command-log.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        break
+      case VerbosityLevel.Normal:
+        this.error = error
+        this.log = log
+        this.info = info
+        this.divider = divider
+        this.dim = emptyFunction
+        break
+      default:
+        // quite
```
Comment (vojtechsimetka) [inline src/command/root-command/command-log.ts:57]:
```suggestion
        // quiet
```
or maybe even better `silence`

**Comment by agazso** [comment]
I tried out with the different verbosity levels:

Screenshot 2021-02-15 at 15 27 49

My suggestions are the following:
- In level 0 only print only the bzz link (e.g. `http://localhost:1633/bzz/<hash>`>
- In level 1 omit the dark grey text (`Data has been synced...`), this could be the default
- Only print the dark grey text in level 2 (verbose mode) but maybe change the color to something more readable.

File: src/command/upload.ts (hunk 2 above)
Comment (nugaon) [reply src/command/upload.ts:8]: It uses "bold" and "green" still, the commit suggestion is wrong

**Review by agazso — APPROVED** [review]

File: src/command/root-command/command-log.ts (hunk 3 above)
Comment (nugaon) [reply src/command/root-command/command-log.ts:64]: silent is identical to "Quiet". If someone wants totally silent output, the output can be piped to `/dev/null`. it is nice to print error message regardless of the given verbosity level on error, especially if the cli exits with error code anyway.
The last that you mentioned is too rigid to implement. Other message types can come into the picture.

**Comment by nugaon** [comment]
> In level 0 only print only the bzz link (e.g. http://localhost:1633/bzz/<hash>>

I think the hash-only could be more useful as it does not contain arbitrary information and also there would be other use-cases when the gateway link does not matter.

> In level 1 omit the dark grey text (Data has been synced...), this could be the default

I think it is not relevant information, the point of "Normal" verbosity mode is with minimal formatting (and with deterministic row count) I get the result of the operation, otherwise some error happened.

> Only print the dark grey text in level 2 (verbose mode) but maybe change the color to something more readable.

The point of the `dim` messages to be hidden in the context, these are only for debugging information and tracking of the progress.
`dim` messages are only visible in verbosity level 2. it was a typeo that I left in the code, thereby it showed also in verbosity level 1, @vojtechsimetka  already gave change suggestion for that to fix.

**Review by vojtechsimetka — APPROVED** [review]

**Pull request merged**